src/mongocxx/database.hpp

```diff
@@ -248,6 +248,7 @@ class MONGOCXX_API database {
 
     class MONGOCXX_PRIVATE impl;
     std::unique_ptr<impl> _impl;
+    const std::unique_ptr<impl>& get_impl() const;
```

Comment:
Having the const and non-const getters seems fine.

The _impl field is currently a std::unique_ptr<impl>. I'm not sure why--perhaps so we don't have to destroy it in the destructor? Do you think I should change it to an impl*?